Definition of coordinate systems : add formulas (#60)
ices-eg/wg_WGFAST: SONAR-netCDF4/docs/crr341.adoc

x\\
y\\
z
\end{bmatrix}_{B} = R_{A->B} \cdot \begin{bmatrix}
x\\
y\\
z
\end{bmatrix}_{A} +
T_{A->B}
\end{equation}
++++

> Rotation is specified via three rotations variables and can be implemented via matrix multiplication. As an example, the matrix that will convert a coordinate from coordinate system PCS to coordinate system SCS is defined by:

Reviewer: ```suggestion
Rotation is specified via three rotation variables and can be implemented via matrix multiplication. As an example, the matrix that will convert a coordinate from coordinate system PCS to coordinate system SCS is defined by:
```

[stem]
++++
\begin{equation}
R_{PCS->SCS} = R_z(h) \cdot R_y(p) \cdot R_z(r)
\end{equation}
++++

where latexmath:[R_z(h)] is the rotation about the _z_-axis by _h_ degrees, etc. The rotation symbols are chosen to reflect their common nautical names: _h_ for heading, _p_ for pitch, and _r_ for roll and used thus:

[stem]
++++
\begin{eqnarray}

R_z(h) =
\begin{bmatrix}
\cos(h) & -\sin(h) & 0 \\
\sin(h) & \cos(h) & 0 \\
0 & 0 & 1
\end{bmatrix} \\

R_y(p) =
\begin{bmatrix}
\cos(p) & 0 & \sin(p) \\
0 & 1 & 0 \\
-\sin(p) & 0 & cos(p)
\end{bmatrix} \\

R_x(r) =
\begin{bmatrix}
1 & 0 & 0 \\
0 & \cos(r) & -\sin(r) \\
0 & \sin(r) & \cos(r)
\end{bmatrix}

\end{eqnarray}
++++

noting that the order of the matrices is important and should follow this order.

To be clear, the right-hand rule when applied to roll, pitch, and heave means that:

* looking along the positive _x_-axis, a positive rotation (roll) is clockwise (to starboard),
* looking along the positive _y_-axis, a positive rotation (pitch) is clockwise (bow up),
* looking along the positive _z_-axis, a positive rotation (heading) is clockwise (to starboard).

> The orientation of the platform is represented using the _z_–_y_’–_x_” Tait-Bryan intrinsic rotation convention (https://en.wikipedia.org/wiki/Euler_angles[en.wikipedia.org/wiki/Euler_angles]), corresponding to heading, pitch, and roll, respectively. Intrinsic means that rotations about the _y_-axis are measured after any rotation about the _z_-axis, and rotations about the _x_-axis are measured after rotations about the _y_-axis (for comparison, extrinsic angles are applied relative to a fixed-platform orientation). This is the most used rotation convention in the maritime field, and the main effect is that roll is measured relative to the plane tilted by the pitch angle.

Reviewer: I thought we had moved to extrinsic rotations? Or perhaps I am confused...

Author: if I trust wikipedia, it is written that "the intrinsic rotations are known as: yaw, pitch and roll"

Author: If I trust wikipedia, it is written that "the intrinsic rotations are known as: yaw, pitch and roll"
So I guess we are intrinsic

Reviewer: I understand it (again) now! So, yes, intrinsic

* Positionning is computed with respect to rx_beam only, meaning we do not make any computation for intersection between tx_beam and rx_beam and taking into account for the ship displacement between emission and reception

Two main categories of system are considered as case studies : system with beam stabilization and pointing angle relative to SCS (typically ME70) and system where beam are not stabilised and
beam pointing angles relative to TCS.

More complex equations and algorithm for more precise positioning or for other sensors are out of scope of this section.

Given an echo _E(beam,i)_ where _i_ is the sample number (from zero to one less than the number of samples) and _beam_ the receiving beam considered, several values can be associated with this echo :

* A time stamp given by
[stem]
++++
\begin{equation}
t_i=ping\_time+\frac{i\cdot sample\_interval + blanking\_interval- sample\_time\_offset}{2}
\end{equation}
++++

* Its range stem:[r_i] given by equation \eqref{eq:timeToRange}

==== Beams are stabilized by the sonar and beam pointing angles are relative to SCS

> When beam are stabilized by the sonar their angles are relative to SCS, the sonar will change beam angles relatively to TCS dynamically to follow ship movement and maintain beam angle constants with respect to the SCS coordinate system.

Reviewer: ```suggestion
When beam are stabilized by the sonar their angles are relative to SCS the sonar will dynamically change beam angles relative to TCS to follow platform movement and maintain a constant beam angle with respect to the SCS.
```

> The coordinates system change from TCS to SCS is composed by a translation from transducer origin to platform origin taking into account lever arms, and a rotation taking into account heading

Reviewer: ```suggestion
The change from TCS to SCS is composed of a translation from transducer origin to platform origin, taking into account lever arms, and a rotation taking into account platform heading
```

\cdot
\begin{bmatrix}
0\\
0\\
r_i
\end{bmatrix}
\end{equation}
++++

with latexmath:[\phi] and latexmath:[\theta] the split-aperture coordinates as defined in <<Split-aperture coordinates>>:

and
[stem]
++++
\begin{equation}
R_{BCS->TCS} =
\begin{bmatrix}
\cos(\alpha) & -\sin(\alpha) & 0 \\
\sin(\alpha) & \cos(\alpha) & 0 \\
0 & 0 & 1
\end{bmatrix}
\cdot
\begin{bmatrix}
\cos(\beta) & 0 & \sin(\beta) \\
0 & 1 & 0 \\
-\sin(\beta) & 0 & cos(\beta)
\end{bmatrix}
\cdot
\begin{bmatrix}
1 & 0 & 0 \\
0 & \cos(\gamma) & -\sin(\gamma) \\
0 & \sin(\gamma) & \cos(\gamma)
\end{bmatrix}
\end{equation}
++++

where stem:[\alpha,\beta,\gamma] are the beam pointings angles as given in the rx_beam_rotation_alpha, rx_beam_rotation_beta, rx_beam_rotation_gamma variables of the beam mode subgroups of the sonar group (see <<sonarBeammodeTable>>)

and:

[stem]
++++
\begin{equation}
R_{PCS/SCS}(t_i) = R_{z}(h(t_i)) \cdot R_{y}(p(t_i)) \cdot R_{x}(r(t_i))
\end{equation}
++++

where stem:[h,p,r] are the heading, pitch and roll as given at the time of the ping in the platform_heading, platform_pitch, platform_roll variables of the beam mode subgroups of the sonar group (see <<sonarBeammodeTable>>).

> (_Those variables can also be defined at a higher frame rate in the platform group  (see <<platformGroupTable>>)_)

Reviewer: ```suggestion
(_Those variables can also be defined at a higher sample rate in the platform group  (see <<platformGroupTable>>)_)
```

and:

[stem]
++++
\begin{equation}
T_{TCS->PCS} = \begin{bmatrix}
transducer\_offset\_x \\
transducer\_offset\_y \\
transducer\_offset\_z
\end{bmatrix}
\end{equation}
++++
where transducer_offset_x, transducer_offset_y and transducer_offset_z are the installation offsets of the transducer associated
with the echo, defined in the platform group  (see <<platformGroupTable>>)

==== Beams are not stabilized by the sonar and beam pointing angles are relative to TCS

> When beam are not stabilized by the sonar, the beam angles  are given as relative to the TCS and

Reviewer: ```suggestion
When beam are not stabilized by the sonar, the beam angles  are given relative to the TCS and
```

> the sonar will not change the beam angles to follow dynamically ship movement during an emission reception cycle.

Reviewer: ```suggestion
the sonar will not dynamically change the beam angles to follow the platform movement during a transmit/receive cycle.
```

\end{equation}
++++

with transducer_rotation_x, transducer_rotation_y and transducer_rotation_z,  the installation angles of the transducer associated
with the echo, defined in the platform group (see <<platformGroupTable>>)
(Note that calibration angles are supposed to be integrated in these variables)

and:

[stem]
++++
\begin{equation}
T_{TCS->PCS} = \begin{bmatrix}
transducer\_offset\_x \\
transducer\_offset\_y \\
transducer\_offset\_z
\end{bmatrix}
\end{equation}
++++
with transducer_offset_x, transducer_offset_y and transducer_offset_z the installation offsets of the transducer associated
with the echo, defined in the platform group (see <<platformGroupTable>>)

and:

[stem]
++++
\begin{equation}
R_{PCS->SCS}(t_i)  = R_{z}(h(t_i)) \cdot R_{y}(p(t_i)) \cdot R_{x}(r(t_i))
\end{equation}
++++
where stem:[h,p,r] are the heading, pitch and roll as given in the platform_heading, platform_pitch, platform_roll variables of the sonar group (see <<sonarBeammodeTable>>) or platform group (see <<platformGroupTable>>)

=== Equations for positioning echoes in a Geographical Coordinate System

> The Surface Coordinate system is a coordinate system oriented with platform heave with origin the platform origin.

Reviewer: Is this what you mean:?
```suggestion
The Surface Coordinate system is a coordinate system oriented with platform heave at the origin of the platform origin.
```

> In order to define an absolute positioning, the coordinate should be converted to a Terrestrial Geographic System coordinate and take into account for heave, draft, tide and so on.

Reviewer: ```suggestion
In order to define an absolute position, the coordinate should be converted to a Terrestrial Geographic System coordinate and take into account heave, draft, tide and so on.
```

> The Terrestrial Geographic System coordinate will use latitude and longitude to define it positions with respect to Earth, and elevation will be defined with respect to a absolute surface reference.

Reviewer: ```suggestion
The Terrestrial Geographic System coordinate will use latitude and longitude to define its' position with respect to Earth and elevation will be defined with respect to a absolute surface reference.
```

> The surface reference could be the WGS84 ellipsoid, or surface like MSL (Mean Sea Level) or LAT (Lowest astronomical tide)

Reviewer: ```suggestion
The absolute surface reference could be the WGS84 ellipsoid, or a surface such as MSL (Mean Sea Level) or LAT (Lowest Astronomical Tide)
```

> Switching from SCS system to TGS is done by applying a vertical translation from the platform origin to the surface reference of the TGS

Reviewer: ```suggestion
Switching from SCS system to TGS is achieved by applying a vertical translation from the platform origin to the surface reference of the TGS
```

> The vertical translation at each time is a vector is obtained by adding the variables :

Reviewer: ```suggestion
The vertical translation at each time is a vector obtained by adding the variables :
```
